pkg/sqlite/image.go

```diff
@@ -211,7 +214,7 @@ func (i *ImageLoader) loadOperatorBundle(manifest registry.PackageManifest, bund
 		return nil
 	}
 
-	if err := i.store.AddBundlePackageChannels(manifest, bundle); err != nil {
+	if err := i.store.AddBundlePackageChannels(manifest, bundle, i.mode); err != nil {
```

Comment:
I sort of expected to see different function calls here based on the mode - seems like there will be a ton of conditional logic based on the mode in this otherwise.

Reply:
Theres very little mixed logic, I just rely on the bundle add as is and then branch to a totally different logic path when processing channel data based on mode. I think like above it we wanted to do clean totally separate implementations per mode it would require some extensive refactoring of the calls to the data layer -- which I was explicitly attempting to avoid based on feedback from earlier this week. I do agree it would be cleaner though